src/components/volunteer/VolunteerEventRecord.tsx

```diff
@@ -73,6 +75,47 @@ export const VolunteerEventRecord = ({
       timeTo: momentFormat(endDatetime, 'HH:mm')
     }
   ];
+  const shareContent = route.params?.shareContent || undefined;
+
+  const [isMe, setIsMe] = useState<boolean>();
+
+  const checkIfMe = useCallback(async () => {
+    const { currentUserId } = await volunteerUserData();
+
+    !!currentUserId && setIsMe(currentUserId == content?.metadata?.created_by?.id);
+  }, [data]);
+
+  useEffect(() => {
+    checkIfMe();
+  }, [checkIfMe]);
+
+  useLayoutEffect(() => {
+    if (isMe) {
+      navigation.setOptions({
+        headerRight: () =>
+          isMe && (
+            <WrapperRow style={styles.headerRight}>
+              <TouchableOpacity
+                onPress={() =>
+                  navigation?.navigate(ScreenName.VolunteerForm, {
+                    query: QUERY_TYPES.VOLUNTEER.CALENDAR,
+                    calendarData: data,
+                    groupId: content?.metadata?.contentcontainer_id
+                  })
+                }
+              >
+                <Icon.NamedIcon name="settings" color="white" style={styles.icon} />
```

Comment:
`white` should com from colors. i will change that.
in addition i will replace the settings icon with `Icon.EditSetting`, that we use in other places:

<img width="373" alt="image" src="https://user-images.githubusercontent.com/1942953/199027118-10c56d73-5057-49da-8321-d69cc44501ca.png">

Reply:
01b627a